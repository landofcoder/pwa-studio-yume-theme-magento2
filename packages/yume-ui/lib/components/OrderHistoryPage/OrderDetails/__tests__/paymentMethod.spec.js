import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import PaymentMethod from '../paymentMethod';

jest.mock('react-intl', () => ({
    FormattedMessage: props => (
        <div componentName="Formatted Message Component" {...props} />
    )
}));

const defaultProps = {
    data: [
        {
            type: 'Credit Card',
            additional_data: [
                {
                    name: 'card_type',
                    value: 'Visa'
                },
                {
                    name: 'last_four',
                    value: '1234'
                }
            ]
        }
    ]
};

test('should render properly', () => {
    const tree = createTestInstance(<PaymentMethod {...defaultProps} />);

    expect(tree.toJSON()).toMatchSnapshot();
});
