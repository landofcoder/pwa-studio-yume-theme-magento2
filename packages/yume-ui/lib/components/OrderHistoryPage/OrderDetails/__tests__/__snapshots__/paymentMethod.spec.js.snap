// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<div>
  <div>
    <div
      componentName="Formatted Message Component"
      defaultMessage="Payment Method"
      id="orderDetails.paymentMethodLabel"
    />
  </div>
  <div>
    Credit Card - Visa
  </div>
  <div>
    1234
  </div>
</div>
`;
